### DISCUSSION TITLE
is there a way to use pocketbase with authentication that happens outside of pocketbase?

### DISCUSSION THREAD

**Discussion by gokulgseven** [pocketbase/pocketbase#3460] [Q&A]
I have a frontend in svelte, backend in express.js and database in Mongodb . 
currently all my users are managed in a user collection inside express and Mongodb. if a uses send is username and password , he gets back a jwt. 

what I want to do is use this authentication to, provide access to user to the pocket base database too. 

basically, I want to authenticate a user from outside pocketable and provide him access to the pocket base Apis, along with the row level security that pocketable offers. (all this while not creating any users in the pocket base database,)

is something like this possible???

**Comment by ganigeorgiev** [comment]
There is not enough details to understand the full picture of your use case and why you would want to do it, but unless you duplicate the users from your MongoDB database in the related PocketBase collection I don't think there is a way to use the `@request.auth.*` filter rule.

And even if you decide to go through this route, you still will have to separately authenticate the user in PocketBase (aka. sending `Authorization: TOKEN` header with each request).

**Reply by gokulgseven** [reply]
My use case.
I have a express.js backend, that takes care of quite a lot of the business logic that my app needs. It does that with a mongodb database and Api endpoints.

now I was thinking of making use of Pocketbase to take care of a part of the business logic that is left.

I am following a microservice architecture for my backend, there is an Api gateway and then multiple express.js apps that take care of separate business logic, each connected to its own database.

there is a main auth (express.js) application, and every request flows through it with the userId, companyId, username, role etc added to the header.

so if i make a request to the endpoint /item/create the request will first go to the auth (express.js) the user will get authenticated based on jwt, the userData will be added to the headers, and the request will go to the items app( express.js) which only gets a request if the user is authenticated.

i am trying to use pocketbase in replacement of one of my express.js apps  

do you think something like this is possible ???

i am able to send the Authorization token in the header, but for this i have to store the passwords in the mongodb database and login the user from inside the backend. (what i dont want to do is maintain a copy of the users) 

If we know the jWT secret and the structure that pocketbase uses, i might be able to create the token on my own , with some predefined userdata , and call pocketbasess api with that. 
but then i would have to be able to use the data provided in the jwt inside the rls of pocketbase .

**Reply by ganigeorgiev** [reply]
In this case I think you'll be better with using your existing setup otherwise I feel that it will add a lot of complexity in your code without much benefit in return.

You can interact with PocketBase APIs as admin, aka. as a "internal database" behind your node project, but if you want to use the `@request.auth.*` filter checks you'll need to duplicate your users in PocketBase because the rules are translated to raw SQL statements, aka. the users need to exist in the underlying PocketBase SQLite db file, we can't perform queries against the MongoDB database.
